taxcalc/parameters.py

```diff
@@ -1,170 +1,496 @@
-import numpy as np
+"""
+OSPC Tax-Calculator federal tax policy Parameters class.
+"""
+# CODING-STYLE CHECKS:
+# pep8 --ignore=E402 parameters.py
+# pylint --disable=locally-disabled parameters.py
+
 from .utils import expand_array
 import os
 import json
-from pkg_resources import resource_stream, Requirement
-
-DEFAULT_START_YEAR = 2013
 
 
 class Parameters(object):
-
-    CUR_PATH = os.path.abspath(os.path.dirname(__file__))
-    PARAM_FILENAME = "params.json"
-    params_path = os.path.join(CUR_PATH, PARAM_FILENAME)
-
-    # Mapping of year to inflation rate
+    """
+    Constructor for the federal tax policy parameters class.
+
+    Parameters
+    ----------
+    parameter_dict: dictionary of PARAM:DESCRIPTION pairs
+        dictionary of policy parameters; if None, default policy
+        parameters are read from the params.json file.
+
+    start_year: integer
+        first calendar year for historical policy parameters.
+
+    num_years: integer
+        number of calendar years for which to specify policy parameter
+        values beginning with start_year.
+
+    inflation_rates: dictionary of YEAR:RATE pairs
+        variable inflation rates used to project future policy parameter
+        values; if None, default inflation rates (specified below) are used.
+
+    Raises
+    ------
+    ValueError:
+        if num_years is less than one.
+        if len(inflation_rates) is not equal to num_years.
+        if min(inflation_rates.keys()) is not equal to start_year.
+
+    Returns
+    -------
+    class instance: Parameters
+    """
+
+    PARAMS_FILENAME = 'params.json'
```

Comment:
@talumbau, what are the tradeoffs between module scope and class scope for these variables?